File: smartmoneyserver/src/main/java/fpt/aptech/server/enums/date/DateRangeMode.java
package fpt.aptech.server.enums.date;

/**
 * Chế độ xem thời gian mà Frontend yêu cầu "nhà máy" sản xuất.
 * <p>
 * Dùng làm tham số `mode` cho API {@code /api/utils/date-ranges} để
 * quyết định xem nên tạo danh sách theo Ngày, Tuần, Tháng, Quý, hay Năm.
 */
public enum DateRangeMode {
    DAILY,      // Theo ngày
    WEEKLY,     // Theo tuần
    MONTHLY,    // Theo tháng
    QUARTERLY,  // Theo quý
    YEARLY;     // Theo năm
}

File: APP MOBILE/smart_money/lib/core/enums/date_range_mode.dart
/// Chế độ xem thời gian: theo ngày, tuần, tháng, quý, năm
enum DateRangeMode {
  daily,
  weekly,
  monthly,
  quarterly,
  yearly;

  String get displayName {
    switch (this) {
      case DateRangeMode.daily:
        return 'Daily';
      case DateRangeMode.weekly:
        return 'Weekly';
      case DateRangeMode.monthly:
        return 'Monthly';
      case DateRangeMode.quarterly:
        return 'Quarterly';
      case DateRangeMode.yearly:
        return 'Yearly';
    }
  }
}

